File: ios/Releaf/Features/Auth/SignInScreen.swift
/*
 * SignInScreen.swift
 * Signed-out landing page. Four sections stacked vertically:
 *   - Top: brand lockup (leaf + wordmark + tagline + hero headline +
 *          one-line value prop).
 *   - BrandProductHero: notebook + phone-scan mockups inside a card.
 *   - BrandLoopSummary: three Write / Erase / Repeat info rows.
 *   - Bottom: sign-in button + Drive-scope reassurance.
 *
 * Top padding is `AppSpacing.s10 + AppSpacing.s6` (64pt) so the hero breathes
 * under the status bar before the brand lockup begins. The page is now
 * back to vertical scrolling because the BrandLoopSummary pushes total
 * content past a single viewport on most phones; without scroll the
 * sign-in button would be clipped on smaller devices.
 *
 * To ship a real sign-in, swap the stub for a GoogleSignIn-iOS-backed
 * client inside AuthStore. The UI here doesn't change.
 */

import SwiftUI
import ReleafDesignSystem
import ReleafData

public struct SignInScreen: View {
    @EnvironmentObject private var authStore: AuthStore
    private let brandGreen = Color(hex: 0x0B3F26)
    private let brandGreenLight = Color(hex: 0x7AA874)
    private let brandGreenDeep = Color(hex: 0x0B4328)

    public init() {}

    public var body: some View {
        ZStack {
            DotGridBackground().ignoresSafeArea()
            ScrollView {
                VStack(alignment: .leading, spacing: AppSpacing.s5) {
                    header
                    BrandProductHero()
                    BrandLoopSummary()
                    actionArea
                }
                .padding(.horizontal, AppSpacing.s5)
                .padding(.top, AppSpacing.s10 + AppSpacing.s6)
                .padding(.bottom, AppSpacing.s6)
                .frame(maxWidth: 540)
                .frame(maxWidth: .infinity)
            }
        }
    }

    private var header: some View {
        VStack(alignment: .leading, spacing: AppSpacing.s3) {
            ReleafLogoRow(
                size: .lg,
                leafGradientStart: brandGreenLight,
                leafGradientEnd: brandGreenDeep,
                wordmarkColor: brandGreen
            )

            Text("WRITE. ERASE. REPEAT.")
                .font(.system(size: 12, weight: .bold, design: .default))
                .tracking(2)
                .foregroundStyle(brandGreen)

            Text("Your ideas deserve\nmore than one life.")
                .font(.system(size: 28, weight: .regular, design: .serif))
                .foregroundStyle(AppColors.textPrimary)
                .lineSpacing(2)
                .fixedSize(horizontal: false, vertical: true)

            Text("Reusable notebook + smart app companion.")
                .font(AppText.body)
                .foregroundStyle(AppColors.textSecondary)
                .fixedSize(horizontal: false, vertical: true)
        }
    }

    private var actionArea: some View {
        VStack(spacing: AppSpacing.s3) {
            AppButton("Sign in with Google") {
                // Real flow when `GIDClientID` is set in Info.plist,
                // otherwise falls back to the stub client path.
                let signIn = GoogleSignInBinding.signInAction(authStore: authStore)
                Task { await signIn() }
            }
            .frame(maxWidth: .infinity)

            if case .failed(let message) = authStore.state {
                Text(message)
                    .font(AppText.meta)
                    .foregroundStyle(AppColors.danger)
            }

            Text("Releaf only sees files it creates in your Drive.")
                .font(AppText.meta)
                .foregroundStyle(AppColors.textTertiary)
                .multilineTextAlignment(.center)
                .frame(maxWidth: .infinity)
        }
    }
}

private struct BrandProductHero: View {
    var body: some View {
        HStack(alignment: .bottom, spacing: AppSpacing.s3) {
            NotebookMockup()
                .frame(maxWidth: .infinity)
                .frame(height: 168)
            PhoneScanMockup()
                .frame(width: 96, height: 160)
        }
        .padding(AppSpacing.s3)
        .background(
            RoundedRectangle(cornerRadius: AppRadius.md, style: .continuous)
                .fill(AppColors.cardSolid)
        )
        .overlay(
            RoundedRectangle(cornerRadius: AppRadius.md, style: .continuous)
                .stroke(AppColors.borderDefault, lineWidth: 1)
        )
        .appShadow(.xs)
        .accessibilityElement(children: .combine)
        .accessibilityLabel("Reusable notebook and Releaf app preview")
    }
}

private struct NotebookMockup: View {
    private let brandGreen = Color(hex: 0x0B3F26)

    var body: some View {
        ZStack(alignment: .topLeading) {
            RoundedRectangle(cornerRadius: 18, style: .continuous)
                .fill(brandGreen)
            VStack(alignment: .leading, spacing: 8) {
                RoundedRectangle(cornerRadius: 10, style: .continuous)
                    .fill(Color(hex: 0xEDE4CF))
                    .frame(height: 110)
                    .overlay(alignment: .topLeading) {
                        VStack(alignment: .leading, spacing: 8) {
                            Text("Plan better tomorrow")
                                .font(.system(size: 11, weight: .bold))
                                .foregroundStyle(brandGreen)
                                .lineLimit(1)
                            ForEach(0..<4, id: \.self) { index in
                                RoundedRectangle(cornerRadius: 2)
                                    .fill(Color(hex: 0xCBBF9E))
                                    .frame(width: index == 2 ? 60 : 82, height: 2.5)
                            }
                        }
                        .padding(11)
                    }

                Text("Reusable notebook")
                    .font(.system(size: 12, weight: .bold))
                    .foregroundStyle(AppColors.onAccent)
            }
            .padding(12)

            ReleafLogo(
                size: 26,
                filled: true,
                fillGradientStart: Color(hex: 0x7AA874),
                fillGradientEnd: brandGreen,
                lineWidth: 1
            )
            .offset(x: 90, y: 100)
        }
    }
}

private struct PhoneScanMockup: View {
    private let brandGreen = Color(hex: 0x0B3F26)

    var body: some View {
        RoundedRectangle(cornerRadius: 20, style: .continuous)
            .fill(Color(hex: 0x101610))
            .overlay {
                VStack(spacing: 7) {
                    Capsule()
                        .fill(Color.black.opacity(0.5))
                        .frame(width: 36, height: 5)
                    VStack(alignment: .leading, spacing: 7) {
                        Text("Project Plan")
                            .font(.system(size: 11, weight: .bold))
                            .foregroundStyle(brandGreen)
                        ForEach(0..<3, id: \.self) { index in
                            HStack(spacing: 6) {
                                Circle()
                                    .fill(brandGreen)
                                    .frame(width: 4, height: 4)
                                RoundedRectangle(cornerRadius: 2)
                                    .fill(Color(hex: 0xD8CBB3))
                                    .frame(width: index == 2 ? 38 : 56, height: 3)
                            }
                        }
                        Spacer(minLength: 0)
                        HStack {
                            Spacer()
                            ReleafLogo(
                                size: 28,
                                filled: true,
                                fillGradientStart: Color(hex: 0xB6D3AA),
                                fillGradientEnd: Color(hex: 0x7AA874),
                                lineWidth: 0.8
                            )
                            .opacity(0.7)
                        }
                    }
                    .padding(10)
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
                    .background(
                        RoundedRectangle(cornerRadius: 14, style: .continuous)
                            .fill(AppColors.canvas)
                    )
                    Text("Scan. Save. Share.")
                        .font(.system(size: 10, weight: .bold))
                        .foregroundStyle(AppColors.onAccent)
                        .padding(.bottom, 4)
                }
                .padding(8)
            }
    }
}

private struct BrandLoopSummary: View {
    private let items: [(String, String, BrandIconKind)] = [
        ("Write",  "Capture thoughts, plans, and pages.",        .write),
        ("Erase",  "Wipe clean with water and start again.",     .erase),
        ("Repeat", "Digitize, search, and share anytime.",       .repeatLoop),
    ]

    var body: some View {
        VStack(alignment: .leading, spacing: AppSpacing.s4) {
            Text("BUILT FOR THE LOOP")
                .font(.system(size: 13, weight: .bold))
                .tracking(1.2)
                .foregroundStyle(Color(hex: 0x0B3F26))
            ForEach(items.indices, id: \.self) { index in
                let item = items[index]
                LoopRow(title: item.0, copy: item.1, icon: item.2)
            }
        }
        .padding(AppSpacing.s4)
        .background(
            RoundedRectangle(cornerRadius: AppRadius.lg, style: .continuous)
                .fill(AppColors.cardSolid)
        )
        .overlay(
            RoundedRectangle(cornerRadius: AppRadius.lg, style: .continuous)
                .stroke(AppColors.borderDefault, lineWidth: 1)
        )
    }
}

// ── Helpers ─────────────────────────────────────────────────────────
// Brand-icon set + LoopRow chrome shared by BrandLoopSummary above.

private enum BrandIconKind { case write, erase, repeatLoop, leaf, phone }

private extension BrandIconKind {
    /// SF Symbol name used by the iOS LoopRow + BrandIcon. Mirrors the
    /// hand-drawn glyph set the Android Canvas BrandIcon emits for
    /// Write / Erase / Repeat / Leaf / Phone. The Releaf "erase" is a
    /// water wipe (cloth + droplets), so the symbol leans on the water
    /// metaphor — `drop.fill` is the closest single SF symbol; we
    /// could swap to a custom Path later to get the cloth + drops
    /// composition seen in the Android Canvas BrandIcon.
    var systemName: String {
        switch self {
        case .write:      return "hand.draw"          // hand-with-pencil
        case .erase:      return "sparkles"           // wipe-and-shine
        case .repeatLoop: return "arrow.triangle.2.circlepath"
        case .leaf:       return "leaf.fill"
        case .phone:      return "iphone"
        }
    }
}

private struct BrandIcon: View {
    let kind: BrandIconKind
    var size: CGFloat = 17
    var body: some View {
        Image(systemName: kind.systemName)
            .font(.system(size: size))
            .foregroundStyle(Color(hex: 0x0B3F26))
    }
}

private struct LoopRow: View {
    let title: String
    let copy: String
    let icon: BrandIconKind

    var body: some View {
        HStack(alignment: .top, spacing: AppSpacing.s3) {
            ZStack {
                Circle().fill(AppColors.greenSoft)
                BrandIcon(kind: icon)
            }
            .frame(width: 42, height: 42)

            VStack(alignment: .leading, spacing: 3) {
                Text(title)
                    .font(.system(size: 15, weight: .bold))
                    .foregroundStyle(Color(hex: 0x0B3F26))
                Text(copy)
                    .font(AppText.meta)
                    .foregroundStyle(AppColors.textSecondary)
            }
        }
    }
}

#Preview {
    SignInScreen()
        .environmentObject(AuthStore(client: StubGoogleAuthClient()))
}

File: apps/releaf/android/app/src/main/java/app/releaf/mobile/features/auth/SignInScreen.kt
/*
 * SignInScreen.kt
 * Signed-out landing page. Four sections stacked vertically:
 *   - Top: brand lockup (leaf + wordmark + tagline + hero headline +
 *          one-line value prop).
 *   - BrandProductHero: notebook + phone-scan mockups inside a card.
 *   - BrandLoopSummary: three Write / Erase / Repeat info rows.
 *   - Bottom: sign-in button + Drive-scope reassurance.
 *
 * Top padding is `AppSpacing.s10 + AppSpacing.s6` (64dp) so the hero breathes under
 * the status bar before the brand lockup begins. The page is now back
 * to vertical scrolling because the BrandLoopSummary pushes total
 * content past a single viewport on most phones; without scroll the
 * sign-in button would be clipped on smaller devices.
 */

package app.releaf.mobile.features.auth

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.Edit
import androidx.compose.material.icons.outlined.Refresh
import androidx.compose.material.icons.outlined.WaterDrop
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.em
import androidx.compose.ui.unit.sp
import app.releaf.mobile.auth.AuthState
import app.releaf.mobile.ui.components.AppButton
import app.releaf.mobile.ui.components.ReleafLogo
import app.releaf.mobile.ui.components.ReleafLogoRow
import app.releaf.mobile.ui.components.ReleafLogoSize
import app.releaf.mobile.ui.theme.AppColors
import app.releaf.mobile.ui.theme.AppSpacing
import app.releaf.mobile.ui.theme.AppTypography

private val BrandGreen = Color(0xFF0B3F26)
private val BrandGreenLight = Color(0xFF7AA874)
private val Paper = Color(0xFFEDE4CF)
private val PaperLine = Color(0xFFCBBF9E)

@Composable
fun SignInScreen(
    state: AuthState,
    onSignIn: () -> Unit,
    modifier: Modifier = Modifier,
) {
    // Canvas + dot-grid come from the parent `ReleafCanvas` wrapper;
    // don't repaint here or it'll cover the texture.
    Box(
        modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
            .padding(horizontal = AppSpacing.s5)
            .padding(top = AppSpacing.s10 + AppSpacing.s6, bottom = AppSpacing.s6),
    ) {
        Column(
            modifier = Modifier.fillMaxWidth(),
            verticalArrangement = Arrangement.spacedBy(AppSpacing.s5),
            horizontalAlignment = Alignment.Start,
        ) {
            LandingHeader()
            BrandProductHero()
            BrandLoopSummary()
            ActionArea(state = state, onSignIn = onSignIn)
        }
    }
}

@Composable
private fun LandingHeader() {
    Column(verticalArrangement = Arrangement.spacedBy(AppSpacing.s3)) {
        ReleafLogoRow(
            size = ReleafLogoSize.Lg,
            leafGradientStart = BrandGreenLight,
            leafGradientEnd = BrandGreen,
            wordmarkColor = BrandGreen,
        )
        Text(
            text = "WRITE. ERASE. REPEAT.",
            style = TextStyle(
                fontFamily = FontFamily.SansSerif,
                fontWeight = FontWeight.Bold,
                fontSize = 12.sp,
                letterSpacing = 0.14.em,
            ),
            color = BrandGreen,
        )
        Text(
            text = "Your ideas deserve\nmore than one life.",
            style = TextStyle(
                fontFamily = FontFamily.Serif,
                fontSize = 28.sp,
                lineHeight = 32.sp,
            ),
            color = AppColors.TextPrimary,
        )
        Text(
            text = "Reusable notebook + smart app companion.",
            style = AppTypography.Body,
            color = AppColors.TextSecondary,
        )
    }
}

@Composable
private fun BrandProductHero() {
    val shape = RoundedCornerShape(20.dp)
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clip(shape)
            .background(AppColors.CardSolid)
            .border(1.dp, AppColors.BorderDefault, shape)
            .padding(AppSpacing.s3),
        verticalAlignment = Alignment.Bottom,
        horizontalArrangement = Arrangement.spacedBy(AppSpacing.s3),
    ) {
        NotebookMockup(
            modifier = Modifier
                .weight(1f)
                .height(168.dp),
        )
        PhoneScanMockup(
            modifier = Modifier
                .width(96.dp)
                .height(160.dp),
        )
    }
}

@Composable
private fun BrandLoopSummary() {
    val shape = RoundedCornerShape(24.dp)
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .clip(shape)
            .background(AppColors.CardSolid)
            .border(1.dp, AppColors.BorderDefault, shape)
            .padding(AppSpacing.s4),
        verticalArrangement = Arrangement.spacedBy(AppSpacing.s4),
    ) {
        Text(
            text = "BUILT FOR THE LOOP",
            style = TextStyle(
                fontSize = 13.sp,
                fontWeight = FontWeight.Bold,
                letterSpacing = 0.08.em,
            ),
            color = BrandGreen,
        )
        LoopRow(
            title = "Write",
            copy = "Capture thoughts, plans, and pages.",
            icon = Icons.Outlined.Edit,
        )
        LoopRow(
            title = "Erase",
            copy = "Wipe clean with water and start again.",
            icon = Icons.Outlined.WaterDrop,
        )
        LoopRow(
            title = "Repeat",
            copy = "Digitize, search, and share anytime.",
            icon = Icons.Outlined.Refresh,
        )
    }
}

@Composable
private fun NotebookMockup(modifier: Modifier = Modifier) {
    Box(
        modifier = modifier
            .clip(RoundedCornerShape(18.dp))
            .background(BrandGreen)
            .padding(12.dp),
    ) {
        Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(110.dp)
                    .clip(RoundedCornerShape(10.dp))
                    .background(Paper)
                    .padding(11.dp),
            ) {
                Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                    Text(
                        text = "Plan better tomorrow",
                        style = TextStyle(
                            fontSize = 11.sp,
                            fontWeight = FontWeight.Bold,
                        ),
                        maxLines = 1,
                        color = BrandGreen,
                    )
                    repeat(4) { index ->
                        Box(
                            Modifier
                                .fillMaxWidth(if (index == 2) 0.7f else 0.94f)
                                .height(2.5.dp)
                                .clip(RoundedCornerShape(2.dp))
                                .background(PaperLine),
                        )
                    }
                }
            }
            Text(
                text = "Reusable notebook",
                style = TextStyle(fontSize = 12.sp, fontWeight = FontWeight.Bold),
                color = AppColors.OnAccent,
            )
        }
        ReleafLogo(
            modifier = Modifier
                .align(Alignment.BottomEnd)
                .padding(bottom = 30.dp),
            size = 26.dp,
            filled = true,
            fillGradientStart = BrandGreenLight,
            fillGradientEnd = BrandGreen,
            strokeWidth = 1.dp,
        )
    }
}

@Composable
private fun PhoneScanMockup(modifier: Modifier = Modifier) {
    Column(
        modifier = modifier
            .clip(RoundedCornerShape(20.dp))
            .background(Color(0xFF101610))
            .padding(8.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(7.dp),
    ) {
        Box(
            modifier = Modifier
                .width(36.dp)
                .height(5.dp)
                .clip(RoundedCornerShape(99.dp))
                .background(Color.Black.copy(alpha = 0.5f)),
        )
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .weight(1f)
                .clip(RoundedCornerShape(14.dp))
                .background(AppColors.Canvas)
                .padding(10.dp),
        ) {
            Column(verticalArrangement = Arrangement.spacedBy(7.dp)) {
                Text(
                    text = "Project Plan",
                    style = TextStyle(fontSize = 11.sp, fontWeight = FontWeight.Bold),
                    color = BrandGreen,
                )
                repeat(3) { index ->
                    Row(
                        verticalAlignment = Alignment.CenterVertically,
                        horizontalArrangement = Arrangement.spacedBy(6.dp),
                    ) {
                        Box(
                            modifier = Modifier
                                .size(4.dp)
                                .clip(CircleShape)
                                .background(BrandGreen),
                        )
                        Box(
                            modifier = Modifier
                                .width(if (index == 2) 38.dp else 56.dp)
                                .height(3.dp)
                                .clip(RoundedCornerShape(2.dp))
                                .background(Color(0xFFD8CBB3)),
                        )
                    }
                }
            }
            ReleafLogo(
                modifier = Modifier
                    .align(Alignment.BottomEnd),
                size = 28.dp,
                filled = true,
                fillGradientStart = Color(0xFFB6D3AA),
                fillGradientEnd = BrandGreenLight,
                strokeWidth = 1.dp,
            )
        }
        Text(
            text = "Scan. Save. Share.",
            style = TextStyle(fontSize = 10.sp, fontWeight = FontWeight.Bold),
            color = AppColors.OnAccent,
            textAlign = TextAlign.Center,
        )
    }
}

@Composable
private fun ActionArea(state: AuthState, onSignIn: () -> Unit) {
    Column(
        modifier = Modifier.fillMaxWidth(),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(AppSpacing.s3),
    ) {
        AppButton(
            text = "Sign in with Google",
            onClick = onSignIn,
            modifier = Modifier.fillMaxWidth(),
        )

        if (state is AuthState.Failed) {
            Text(
                state.message,
                style = AppTypography.Meta,
                color = AppColors.Danger,
            )
        }

        Text(
            "Releaf only sees files it creates in your Drive.",
            style = AppTypography.Meta,
            color = AppColors.TextTertiary,
            textAlign = TextAlign.Center,
            modifier = Modifier.fillMaxWidth(),
        )
    }
}

// ── LoopRow helper ──────────────────────────────────────────────────
// Used by BrandLoopSummary to render a single Write / Erase / Repeat
// row. Takes a Material `ImageVector` so callers can pass any standard
// icon — we use `Icons.Outlined.Edit / WaterDrop / Refresh` from
// material-icons-extended.

@Composable
private fun LoopRow(title: String, copy: String, icon: ImageVector) {
    Row(
        horizontalArrangement = Arrangement.spacedBy(AppSpacing.s3),
        verticalAlignment = Alignment.Top,
    ) {
        Box(
            modifier = Modifier
                .size(42.dp)
                .clip(CircleShape)
                .background(AppColors.GreenSoft),
            contentAlignment = Alignment.Center,
        ) {
            Icon(
                imageVector = icon,
                contentDescription = null,
                tint = BrandGreen,
                modifier = Modifier.size(23.dp),
            )
        }
        Column(verticalArrangement = Arrangement.spacedBy(3.dp)) {
            Text(
                text = title,
                style = TextStyle(fontSize = 15.sp, fontWeight = FontWeight.Bold),
                color = BrandGreen,
            )
            Text(
                text = copy,
                style = AppTypography.Meta,
                color = AppColors.TextSecondary,
            )
        }
    }
}
